pymc3_hmm/utils.py

```diff
@@ -458,3 +457,62 @@ def plot_predictive_histograms(
             )
 
     return axes
+
+
+def plot_split_timeseries_hists(
+    plot_data: pd.DataFrame,
+    sample_col: str,
+    sample_mean_col: str,
+    sample_mean_state_col: str,
+    observed_col: Union[str, None] = None,
+):  # pragma: no cover
+    from matplotlib.ticker import MaxNLocator
+
+    def plot_fn(ax, data, **kwargs):
+        if observed_col is not None:
+            ax.plot(
+                data[observed_col],
+                label=observed_col,
+                alpha=0.7,
+                color="red",
+                linewidth=0.8,
+                drawstyle="steps",
+            )
+            # make sample mean less visible
+            alpha, color = 0.3, "blue"
+        else:
+            # make sample mean more visible
+            alpha, color = 0.7, "darkorange"
+
+        ax.plot(
+            data[sample_mean_col],
+            label=sample_mean_col,
+            alpha=alpha,
+            color=color,
+            linewidth=0.8,
+            drawstyle="steps",
+        )
+
+        plot_predictive_histograms(data[sample_col], axes=ax)
+
+    axes_split_data = plot_split_timeseries(
+        plot_data,
+        split_max=10,
+        twin_column_name=sample_mean_state_col,
+        twin_plot_kwargs={
+            "color": "green",
+            "drawstyle": "steps",
+            "linestyle": "--",
+            "alpha": 0.4,
+        },
+        figsize=(15, 30),
+        plot_fn=plot_fn,
+    )
+
+    for (_, twin_ax), _ in axes_split_data:
+        _ = twin_ax.set_ylabel(
+            sample_mean_state_col, color=twin_ax.get_lines()[0].get_color()
+        )
+        _ = twin_ax.yaxis.set_major_locator(MaxNLocator(integer=True))
+
+    plt.tight_layout()
```

Comment:
These are the kinds of customizations we need to leave to the caller (especially the twin axes).  We can provide default values for some of these parameters, but&mdash;at the very least&mdash;we need to provide keyword argument pass-throughs to the underlying plot functions and return all the relevant axes they generate.

Reply:
Passing the `plot_split_timeseries` parameters to `**split_ts_kwargs`, apart from `twin_column_name ` as a stand-alone parameter. Returning `axes_split_data` at the end.